h2quic/response_writer.go

```diff
@@ -86,12 +98,129 @@ func (w *responseWriter) Flush() {}
 // TODO: Implement a functional CloseNotify method.
 func (w *responseWriter) CloseNotify() <-chan bool { return make(<-chan bool) }
 
+// Should do:
+// - construct a valid HTTP2 header for a request for the file to push.
+// - open a new stream (server side)
+// - send a PUSH_PROMISE containing the streamID of the new stream and the HTTP2 header
+// - use the header to create a http request and use it in ServeHTTP(w ResponseWriter, r *Request) to serve the file to push.
+// TODO (a lot of these are taken from golang.org/x/net/http2/server.go: push(target string, opts pushOptions) )
+// - check for recursive pushes: "PUSH_PROMISE frames MUST only be sent on a peer-initiated stream."
+func (w *responseWriter) Push(target string, opts *http.PushOptions) error {
+	if w.headerStream.StreamID()%2 == 0 {
```

Comment:
It makes sense in H2, since the push promise there is sent on the request stream. In QUIC we don't need this, since we're sending the promise stream 3.